src/clarity/debug/dap/mod.rs

```diff
@@ -0,0 +1,1146 @@
+            category: Some(Category::Console),
+            output: message.into(),
+            group: None,
+            variables_reference: None,
+            source: None,
+            line: None,
+            column: None,
+            data: None,
+        }));
+    }
+
+    pub fn stdout<S: Into<String>>(&mut self, message: S) {
+        self.send_event(EventBody::Output(OutputEvent {
+            category: Some(Category::Stdout),
+            output: message.into(),
+            group: None,
+            variables_reference: None,
+            source: None,
+            line: None,
+            column: None,
+            data: None,
+        }));
+    }
+
+    pub fn stderr<S: Into<String>>(&mut self, message: S) {
+        self.send_event(EventBody::Output(OutputEvent {
+            category: Some(Category::Stderr),
+            output: message.into(),
+            group: None,
+            variables_reference: None,
+            source: None,
+            line: None,
+            column: None,
+            data: None,
+        }));
+    }
+
+    fn launch(&mut self, seq: i64, arguments: LaunchRequestArguments) -> bool {
+        // Verify that the manifest and expression were specified
+        let manifest = match arguments.manifest {
+            Some(manifest) => manifest,
+            None => {
+                self.send_response(Response {
+                    request_seq: seq,
+                    success: false,
+                    message: Some("manifest must be specified".to_string()),
+                    body: None,
+                });
+                return false;
+            }
+        };
+        let expression = match arguments.expression {
+            Some(expression) => expression,
+            None => {
+                self.send_response(Response {
+                    request_seq: seq,
+                    success: false,
+                    message: Some("expression to debug must be specified".to_string()),
+                    body: None,
+                });
+                return false;
+            }
+        };
+
+        let contract_id = QualifiedContractIdentifier::transient();
+        self.state = Some(DebugState::new(&contract_id, &expression));
+        self.launched = Some((manifest, expression));
+
+        self.launch_seq = seq;
+
+        false
+    }
+
+    fn configuration_done(&mut self, seq: i64) -> bool {
+        self.send_response(Response {
+            request_seq: seq,
+            success: true,
+            message: None,
+            body: Some(ResponseBody::ConfigurationDone),
+        });
+
+        // Now that configuration is done, we can respond to the launch
+        self.send_response(Response {
+            request_seq: seq,
+            success: true,
+            message: None,
+            body: Some(ResponseBody::Launch),
+        });
+
+        false
+    }
+
+    fn setBreakpoints(&mut self, seq: i64, arguments: SetBreakpointsArguments) -> bool {
```

Comment:
```suggestion
    fn setBreakpoints(&mut self, seq: i64, arguments: SetBreakpointsArguments) -> bool {
    fn set_breakpoints(&mut self, seq: i64, arguments: SetBreakpointsArguments) -> bool {
```

Reply:
Oops 😅. My brain is not fully in Rust mode sometimes.